resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -47,120 +47,32 @@ package object config extends Logging {
-        " mounting a secret from the submitting machine into the pod, and hence this" +
-        " configuration is marked as internal, but this can also be set manually to" +
-        " use a key file that is mounted into the driver pod via other means.")
-      .internal()
-      .stringConf
-      .createOptional
-
-  private[spark] val KUBERNETES_DRIVER_MOUNTED_CLIENT_CERT_FILE =
-    ConfigBuilder(s"$APISERVER_DRIVER_CONF_PREFIX.mounted.clientCertFile")
-      .doc("Path on the driver pod's disk containing the client cert file to use when" +
-        " authenticating against Kubernetes. Typically this is configured by spark-submit from" +
-        " mounting a secret from the submitting machine into the pod, and hence this" +
-        " configuration is marked as internal, but this can also be set manually to" +
-        " use a certificate that is mounted into the driver pod via other means.")
-      .internal()
-      .stringConf
-      .createOptional
-
-  private[spark] val KUBERNETES_DRIVER_MOUNTED_OAUTH_TOKEN =
-    ConfigBuilder(s"$APISERVER_DRIVER_CONF_PREFIX.mounted.oauthTokenFile")
-      .doc("Path on the driver pod's disk containing the OAuth token file to use when" +
-        " authenticating against Kubernetes. Typically this is configured by spark-submit from" +
-        " mounting a secret from the submitting machine into the pod, and hence this" +
-        " configuration is marked as internal, but this can also be set manually to" +
-        " use a token that is mounted into the driver pod via other means.")
-      .internal()
-      .stringConf
-      .createOptional
+  private[spark] val APISERVER_AUTH_SUBMISSION_CONF_PREFIX =
+      "spark.kubernetes.authenticate.submission"
+  private[spark] val APISERVER_AUTH_DRIVER_CONF_PREFIX =
+      "spark.kubernetes.authenticate.driver"
+  private[spark] val APISERVER_AUTH_DRIVER_MOUNTED_CONF_PREFIX =
+      "spark.kubernetes.authenticate.driver.mounted"
+  private[spark] val APISERVER_AUTH_RESOURCE_STAGING_SERVER_CONF_PREFIX =
+      "spark.kubernetes.authenticate.resourceStagingServer"
+  private[spark] val APISERVER_AUTH_SHUFFLE_SERVICE_CONF_PREFIX =
+      "spark.kubernetes.authenticate.shuffleService"
+  private[spark] val OAUTH_TOKEN_CONF_SUFFIX = "oauthToken"
+  private[spark] val OAUTH_TOKEN_FILE_CONF_SUFFIX = "oauthTokenFile"
+  private[spark] val CLIENT_KEY_FILE_CONF_SUFFIX = "clientKeyFile"
+  private[spark] val CLIENT_CERT_FILE_CONF_SUFFIX = "clientCertFile"
+  private[spark] val CA_CERT_FILE_CONF_SUFFIX = "caCertFile"
+
+  private[spark] val RESOURCE_STAGING_SERVER_USE_SERVICE_ACCOUNT_CREDENTIALS =
+    ConfigBuilder(
+          s"$APISERVER_AUTH_RESOURCE_STAGING_SERVER_CONF_PREFIX.useServiceAccountCredentials")
+      .doc("Use a service account token and CA certificate in the resource staging server to" +
+        " watch the API server's objects.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
what happens when this is false?  no watching on resource destruction to clean up resources?